File: GameEngine_IOS/GameEngine/Commons/WaveFront/MtlPrefix.swift
/**
 * Set of all the prefixes supported by the material parser
 */
open class MtlPrefix {
    
    /**
     * The prefix of a comment
     */
    open static let COMMENT = "#";
    
    /**
     * The prefix of a new material
     */
    open static let NEW_MATERIAL = "newmtl";
    
    /**
     * The prefix of weight of specular color
     */
    open static let WEIGHT_SPECULAR_COLOR = "Ns";
    
    /**
     * The prefix of specular color
     */
    open static let SPECULAR_COLOR = "Ks";
    
    /**
     * The prefix of ambient color
     */
    open static let AMBIENT_COLOR = "Ka";
    
    /**
     * The prefix of diffuse color
     */
    open static let DIFFUSE_COLOR = "Kd";
    
    /**
     * The prefix of emissive color
     */
    open static let EMISSIVE_COLOR = "Ke";
    
    /**
     * The prefix of optical density
     */
    open static let OPTICAL_DENSITY = "Ni";
    
    /**
     * The prefix of dissolve factor
     */
    open static let DISSOLVE_FACTOR = "d";
    
    /**
     * The prefix of inverse of dissolve factor
     */
    open static let DISSOLVE_FACTOR_INVERTED = "Tr";
    
    /**
     * The prefix of illumination model
     */
    open static let ILLUMINATION_MODEL = "illum";
    
    /**
     * The prefix of transmission filter
     */
    open static let TRANSMISSION_FILTER = "Tf";
    
    /**
     * The prefix of diffuse color texture map
     */
    open static let COLOR_TEXTURE_MAP = "map_Kd";
    
    /**
     * The prefix of specular color texture map
     */
    open static let SPECULAR_COLOR_TEXTURE_MAP = "map_Ks";
    
    /**
     * The prefix of ambient color texture map
     */
    open static let AMBIENT_COLOR_TEXTURE_MAP = "map_Ka";
    
    /**
     * The prefix of bump texture map
     */
    open static let BUMP_TEXTURE_MAP = "map_Bump";
    
    /**
     * The prefix of bump texture map
     */
    open static let BUMP_TEXTURE_MAP_V2 = "bump";
    
    /**
     * The prefix of bump texture map
     */
    open static let BUMP_TEXTURE_MAP_V3 = "map_Disp";
    
    /**
     * The prefix of opacity texture map
     */
    open static let OPACITY_MAP = "map_d";
    
}

File: GameEngine_Android/commons/src/main/java/com/dferreira/commons/waveFront/MtlPrefix.java
package com.dferreira.commons.waveFront;

/**
 * Set of all the prefixes supported by the material parser
 */
public class MtlPrefix {

	/**
	 * The prefix of a comment
	 */
	public final static String COMMENT = "#";
	
	/**
	 * The prefix of a new material
	 */
	public final static String NEW_MATERIAL = "newmtl";

	/**
	 * The prefix of weight of specular color
	 */
	public final static String WEIGHT_SPECULAR_COLOR = "Ns";

	/**
	 * The prefix of specular color
	 */
	public final static String SPECULAR_COLOR = "Ks";

	/**
	 * The prefix of ambient color
	 */
	public final static String AMBIENT_COLOR = "Ka";

	/**
	 * The prefix of diffuse color
	 */
	public final static String DIFFUSE_COLOR = "Kd";

	/**
	 * The prefix of emissive color
	 */
	public final static String EMISSIVE_COLOR = "Ke";

	/**
	 * The prefix of optical density
	 */
	public final static String OPTICAL_DENSITY = "Ni";

	/**
	 * The prefix of dissolve factor
	 */
	public final static String DISSOLVE_FACTOR = "d";

	/**
	 * The prefix of inverse of dissolve factor
	 */
	public final static String DISSOLVE_FACTOR_INVERTED = "Tr";

	/**
	 * The prefix of illumination model
	 */
	public final static String ILLUMINATION_MODEL = "illum";

	/**
	 * The prefix of transmission filter
	 */
	public final static String TRANSMISSION_FILTER = "Tf";

	/**
	 * The prefix of diffuse color texture map
	 */
	public final static String COLOR_TEXTURE_MAP = "map_Kd";

	/**
	 * The prefix of specular color texture map
	 */
	public final static String SPECULAR_COLOR_TEXTURE_MAP = "map_Ks";

	/**
	 * The prefix of ambient color texture map
	 */
	public final static String AMBIENT_COLOR_TEXTURE_MAP = "map_Ka";

	/**
	 * The prefix of bump texture map
	 */
	public final static String BUMP_TEXTURE_MAP = "map_Bump";

	/**
	 * The prefix of bump texture map
	 */
	public final static String BUMP_TEXTURE_MAP_V2 = "bump";

	/**
	 * The prefix of bump texture map
	 */
	public final static String BUMP_TEXTURE_MAP_V3 = "map_Disp";

	/**
	 * The prefix of opacity texture map
	 */
	public final static String OPACITY_MAP = "map_d";

}
